addons/mail/static/src/models/discuss_public_root/discuss_public_root.js

```diff
@@ -0,0 +1,49 @@
+/** @odoo-module **/
+
+import { attr, one2one } from '@mail/model/model_field';
+import { registerNewModel } from '@mail/model/model_core';
+
+function factory(dependencies) {
+
+    class DiscussPublicRoot extends dependencies['mail.model'] {
+
+        /**
+         * Displays the thread view linked to the discuss public root instead of
+         * the welcome view.
+         */
+        showThreadView() {
+            this.update({ activeView: 'thread' });
+        }
+
+    }
+
+    DiscussPublicRoot.fields = {
+        /**
+         * Determines the view (either 'thread' or 'welcome') to be displayed.
+         */
+        activeView: attr({
+            required: true,
+        }),
+        /**
+         * States the thread view linked to this discuss public root.
+         */
+        threadView: one2one('mail.thread_view', {
+            readonly: true,
+            required: true,
+        }),
+        /**
+         * States the welcome view linked to this discuss public root.
+         */
+        welcomeView: one2one('mail.welcome_view', {
+            inverse: 'discussPublicRoot',
+            readonly: true,
+            required: true,
+        }),
+    };
+
+    DiscussPublicRoot.modelName = 'mail.discuss_public_root';
```

Comment:
"root" is related to how it is mounted, not to the view itself, so it's not a good name for it.

In my PR I'm creating this model too and it is "mail.discuss_public_view" to be consistent with the rest.